src/library/autodj/autodjprocessor.cpp

```diff
@@ -859,17 +864,31 @@ void AutoDJProcessor::playerPositionChanged(DeckAttributes* pAttributes,
             double transitionProgress = (thisPlayPosition - thisDeck->fadeBeginPos) /
                     (thisDeck->fadeEndPos - thisDeck->fadeBeginPos);
             double transitionStep = transitionProgress - m_transitionProgress;
+
+            if ((transitionProgress == 0.0) &&
```

Comment:
I am not sure if the == 0.0 state is always reached, because of rounding issues.i think it is more reliable to move this in an else branch of the condition below.

Reply:
Confirmed, this condition is never true and can be removed.

Reply:
After some testing (and an assert), I couldn't find this state being reached either.  So I agree it can be removed.